python/ql/src/experimental/Security/CWE-208/TimingAttack.qll

```diff
@@ -0,0 +1,167 @@
+ *
+ * For example: `request.headers.get("X-Auth-Token")`.
+ */
+abstract class ClientSuppliedsecret extends DataFlow::CallCfgNode { }
+
+private class FlaskClientSuppliedsecret extends ClientSuppliedsecret {
+  FlaskClientSuppliedsecret() {
+    exists(RemoteFlowSource rfs, DataFlow::AttrRead get |
+      rfs.getSourceType() = "flask.request" and this.getFunction() = get
+    |
+      // `get` is a call to request.headers.get or request.headers.get_all or request.headers.getlist
+      // request.headers
+      get.getObject()
+          .(DataFlow::AttrRead)
+          // request
+          .getObject()
+          .getALocalSource() = rfs and
+      get.getAttributeName() in ["get", "get_all", "getlist"] and
+      get.getObject().(DataFlow::AttrRead).getAttributeName() = "headers" and
+      this.getArg(0).asExpr().(StrConst).getText().toLowerCase() = sensitiveheaders()
+    )
+  }
+}
+
+private class DjangoClientSuppliedsecret extends ClientSuppliedsecret {
+  DjangoClientSuppliedsecret() {
+    exists(RemoteFlowSource rfs, DataFlow::AttrRead get |
+      rfs.getSourceType() = "django.http.request.HttpRequest" and this.getFunction() = get
+    |
+      // `get` is a call to request.headers.get or request.META.get
+      // request.headers
+      get.getObject()
+          .(DataFlow::AttrRead)
+          // request
+          .getObject()
+          .getALocalSource() = rfs and
+      get.getAttributeName() = "get" and
+      get.getObject().(DataFlow::AttrRead).getAttributeName() in ["headers", "META"] and
+      this.getArg(0).asExpr().(StrConst).getText().toLowerCase() = sensitiveheaders()
+    )
+  }
+}
+
+private class TornadoClientSuppliedsecret extends ClientSuppliedsecret {
+  TornadoClientSuppliedsecret() {
+    exists(RemoteFlowSource rfs, DataFlow::AttrRead get |
+      rfs.getSourceType() = "tornado.web.RequestHandler" and this.getFunction() = get
+    |
+      // `get` is a call to `rfs`.request.headers.get
+      // `rfs`.request.headers
+      get.getObject()
+          .(DataFlow::AttrRead)
+          // `rfs`.request
+          .getObject()
+          .(DataFlow::AttrRead)
+          // `rfs`
+          .getObject()
+          .getALocalSource() = rfs and
+      get.getAttributeName() in ["get", "get_list"] and
+      get.getObject().(DataFlow::AttrRead).getAttributeName() = "headers" and
+      this.getArg(0).asExpr().(StrConst).getText().toLowerCase() = sensitiveheaders()
+    )
+  }
+}
+
+/** A string for `match` that identifies strings that look like they represent Sensitive Headers. */
+private string sensitiveheaders() {
+  result =
+    [
+      "x-auth-token", "x-csrf-token", "http_x_csrf_token", "x-csrf-param", "x-csrf-header",
+      "http_x_csrf_token", "x-api-key", "authorization", "proxy-authorization"
+    ]
+}
+
+/**
+ * A config that tracks data flow from remote user input to cryptographic operations
+ */
+private class UserInputMsgConfig extends TaintTracking::Configuration {
```

Comment:
```suggestion
class UserInputMsgConfig extends TaintTracking::Configuration {
```
The `TimingAttackAgainstSignature.ql` query was failing because `UserInputMsgConfig` was not defined.

Whenever you make changes, it's always a good idea to check that the code compiles and the tests run correctly before pushing it to the PR.